specification/cognitiveservices/OpenAI.Inference/models/completions/streaming/chat_completions.tsp

```diff
@@ -0,0 +1,126 @@
+import "@typespec/rest";
+import "@typespec/http";
+import "@typespec/versioning";
+
+import "../azure_chat_extensions.tsp";
+import "../common.tsp";
+import "../functions.tsp";
+import "../chat_completions.tsp";
+
+using TypeSpec.Rest;
+using TypeSpec.Http;
+using TypeSpec.Versioning;
+
+namespace Azure.OpenAI;
+
+@doc("A single, role-attributed message within a chat completion interaction.")
+model ChatMessageStreamChunk {
+  @doc("The role associated with this message payload.")
+  @projectedName("json", "role")
+  role: ChatRole;
+
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "we explicitly want a nullable string"
+  @doc("The text associated with this message payload.")
+  @projectedName("json", "content")
+  content?: string; // they key might not be present on a streaming scenario
+  // content is optional in the streaming scenario therefore we define as content?: string
+
+  @doc("""
+    The name of the author of this message. `name` is required if role is `function`, and it should be the name of the
+    function whose response is in the `content`. May contain a-z, A-Z, 0-9, and underscores, with a maximum length of
+    64 characters.
+    """)
+  name?: string;
+
+  @doc("The name and arguments of a function that should be called, as generated by the model.")
+  @added(ServiceApiVersions.v2023_07_01_Preview)
+  @projectedName("json", "function_call")
+  functionCall?: FunctionCall;
```

Comment:
This is another one of those "so close" things between streaming and non-streaming:

- In non-streaming responses ([OpenAI YAML link](https://github.com/openai/openai-openapi/blob/e145786e70bf5fc1bc73c7cd19884f445d52c383/openapi.yaml#L2821)), `name` and `arguments` on `function_call` are both required
- In streaming responses ([OpenAI YAML link](https://github.com/openai/openai-openapi/blob/e145786e70bf5fc1bc73c7cd19884f445d52c383/openapi.yaml#L2843)), `name` and `arguments` on `function_call` are both optional

I think this is an appropriate place to just use an anonymous model in the .tsp, i.e.
```
functionCall?: {
  name?: string,
  arguments?: string
};
```

Reply:
Oops, this crashes the Java emitter 😅 I think that it takes the model name for the file name into which code generation dumps the code. Resulting in something like `<long_path>/.java`